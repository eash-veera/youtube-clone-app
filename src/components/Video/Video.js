import React from 'react';
import './Video.scss';

const BASE_EMBED_URL = 'https://www.youtube.com/embed/';

export function Video (props){
    if( !props.id ){
        return null;
    }
    
    // disabled auto play, coz its annoying while developing !
    // const embedUrl = `${BASE_EMBED_URL}${props.id}?autoplay=1`;

    const embedUrl = `${BASE_EMBED_URL}${props.id}`;

    return (
        <div className='video-container'>
            <div className='video'>
                <iframe className='video-player' src={embedUrl} frameBorder='0' allow='autoplay; encrypted-media' allowFullScreen /> 
            </div>
        </div>
    );
}