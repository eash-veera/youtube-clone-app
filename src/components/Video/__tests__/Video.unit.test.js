import React from 'react';
import { shallow } from 'enzyme';
import { Video } from '../Video';
import expectExport from 'expect';

describe('Video Component', () => {

    test('renders video component correctly' , () => {
        const wrapper = shallow(
            <Video id='HAuXJVI_bUs' />
        );
        
        expect(wrapper).toMatchSnapshot();
    });

    test('renders null if id in video is not specified !' , () => {
        const wrapper = shallow(
            <Video />
        );
        
        expect(wrapper).toMatchSnapshot();
    });
});