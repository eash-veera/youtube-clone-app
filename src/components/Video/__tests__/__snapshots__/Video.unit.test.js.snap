// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video Component renders null if id in video is not specified ! 1`] = `""`;

exports[`Video Component renders video component correctly 1`] = `
<div
  className="video-container"
>
  <div
    className="video"
  >
    <iframe
      allow="autoplay; encrypted-media"
      allowFullScreen={true}
      className="video-player"
      frameBorder="0"
      src="https://www.youtube.com/embed/HAuXJVI_bUs"
    />
  </div>
</div>
`;
